pkg/transport/transport.go

```diff
@@ -30,15 +31,30 @@ type Client interface {
 
 // Transport interface structure
 type Transport struct {
-	tlsOpts   *tlsopts.Options
-	observers []Observer
+	tlsOpts        *tlsopts.Options
+	observers      []Observer
+	requestTimeout time.Duration
+}
+
+// InvokeTimeout enables timeouts for an overly long method call
```

Comment:
I think it will be for all request not only overlay

Reply:
I had "overly long method call" there to mean that the method runs for too long or "over" what it should (not "overlay") but I can take it out so there's no confusion.

Reply:
sorry, no need to change